### PR TITLE
CCMSG-1772 | Remove the logging of sink record in case of data exception.

### PR DESCRIPTION
**Pull request by pbadani**
https://confluentinc.atlassian.net/browse/RCCA-6501

##### Does this solution apply anywhere else?
- [ ] yes
- [x] no

##### Testing done:
- [x] Unit tests
- [ ] Integration tests
- [ ] System tests
- [ ] Manual tests

### CODE REVIEW

File: kafka-connect-s3/src/main/java/io/confluent/connect/s3/format/KeyValueHeaderRecordWriterProvider.java [hunk 1]
```diff
@@ -88,17 +88,13 @@ public void write(SinkRecord sinkRecord) {
 
         // keyWriter != null means writing keys is turned on
         if (keyWriter != null && sinkRecord.key() == null) {
-          throw new DataException(
-              String.format("Key cannot be null for SinkRecord: %s", sinkRecord)
-          );
+          throw new DataException("Key cannot be null for SinkRecord.");
```
Comment (yashmayya) [inline kafka-connect-s3/src/main/java/io/confluent/connect/s3/format/KeyValueHeaderRecordWriterProvider.java:91]: Maybe we can include the topic name for better debuggability?

**Review by yashmayya — APPROVED** [review]
LGTM with minor suggestion.

File: kafka-connect-s3/src/main/java/io/confluent/connect/s3/format/KeyValueHeaderRecordWriterProvider.java (hunk 1 above)
Comment (pbadani) [reply kafka-connect-s3/src/main/java/io/confluent/connect/s3/format/KeyValueHeaderRecordWriterProvider.java:91]: ack

**Review by sudeshwasnik — APPROVED** [review]
lgtm! thanks!

**Pull request merged**